llvmlite/tests/test_binding.py

```diff
@@ -3045,6 +3074,25 @@ def test_instcombine(self):
         self.assertIn("%.3", orig_asm)
         self.assertNotIn("%.3", optimized_asm)
 
+    def test_optnone(self):
+        # Module shouldn't be optimized if the function has `optnone` attached
+        pb = self.pb(speed_level=3, size_level=0)
+        orig_asm_no_optnone = str(asm_alloca_optnone.replace("optnone ", ""))
```

Comment:
A nit, but I'd be inclined to just call this `orig_asm` since it's the same as `orig_asm_optnone` without the `optnone`:

```suggestion
        orig_asm = str(asm_alloca_optnone.replace("optnone ", ""))
```